### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by jlaura**
Updates the README to include discussion of semantic versioning from a user and developer perspective.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Infrastructure change (changes to things like CI or the build system that do not impact users)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by jlaura** [comment]
@AndrewAnnex @rbeyer @KrisBecker Draft text to add to the README desribing semver from user and dev perspectives. All comments and changes solicited.

**Review by rbeyer — APPROVED** [review]

**Review by KrisBecker — COMMENTED** [review]
This is great.

I appreciate the user example that describes what a breaking change looks like pertaining to main application output.

Perhaps a few developer example would be useful. 

Please indicate if these are valid assertions.

Some non-breaking examples:
- Add a new method to a class
- Add a parameter to the end of an existing method argument list that provides a default value.

Some breaking examples:
- Changing the type of a method parameter
- Change the return type of a method
- Inserting a parameter into a method argument list - default provided or not

It may also be worth mentioning (if this statement is true) that while all contributing developers must apply due diligence to identify these conditions before the PR is submitted, the USGS ISIS developers/reviewers/committers are the ultimate authority to identify/determine breaking changes and enforce the rules governing semantic versioning in ISIS.

While you are making changes to this README, can you also update the **Operating System Requirements** with current supported system requirements?

Thanks..

**Comment by rbeyer** [comment]
Kris, I'd agree with your examples, with the addition of this clause to the end of the second non-breaking one: "... provides a default value, which preserves the return value of the method."

However, this README is gargantuan already, and I think additions need to be kept terse, so I'd actually advocate adding these examples to the [More Specific Examples Section in the APIdefinition.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/APIdefinition.md#more-specific-examples) file.

File: README.md [hunk 1]
```diff
@@ -523,3 +524,21 @@ python downloadIsisData --config rclone.conf <mission> $ISISDATA
...
+
+The first two numbers indicate whether the change(s) are breaking or non-breaking. What is a [breaking change](https://github.com/USGS-Astrogeology/ISIS3/wiki/RFC2:-Release-Process#terms)? If a change to [the API](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/APIdefinition.md), defined as programs (e.g., `spiceinit` or `cam2map` or `pds2isis`) and some text output (e.g., CSV output, but not `.txt`), alters the way in which a user calls the program or parses the program output in a way that a scripted solution would fail, that change in breaking. In other words, if a CSV output file removed or renamed a column, that would be breaking. If a CSV file added a new column, that would not be breaking. Likewise, if an application `spiceinit` adds a new arguments, that is non-breaking. If the change removes, reorders, or changes the way the application (CLI) is called, the change is breaking.
```
Comment (AndrewAnnex) [inline README.md]: `...fail, that change in breaking.` suggesting something like `...fail, that would be considered a breaking change.`

**Comment by KrisBecker** [comment]
> Kris, I'd agree with your examples, with the addition of this clause to the end of the second non-breaking one: "... provides a default value, which preserves the return value of the method."
> 

Ok, at the risk this becoming a debate, I think `...preserves the return type and units (if applicable) of the method` is the required criteria. It seems a rather common reason to add an additional parameter is to change the value.

> However, this README is gargantuan already, and I think additions need to be kept terse, so I'd actually advocate adding these examples to the [More Specific Examples Section in the APIdefinition.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/APIdefinition.md#more-specific-examples) file.

That works. Maybe add a link to that document here instead.

**Comment by Kelvinrr** [comment]
@jlaura What is the status of this PR? Been sitting for a while.

**Comment by jlaura** [comment]
I haven't had time to do anything with it / I'm not the software lead and am generally less engaged in ISIS from a management perspective. If someone else wants to take it and close, that would be terrific.

**Review by Kelvinrr — APPROVED** [review]
Addressed the comment and fixed other grammar and spelling mistake, I think this is good.

**Pull request merged**